src/forms/progressive-profiling-popup/index.jsx

```diff
@@ -63,15 +63,16 @@ const ProgressiveProfilingForm = () => {
     }
 
     if (!countryCode) {
+      setFormErrors({ country: formatMessage(messages.progressiveProfilingCountryFieldErrorMessage) });
       return;
     }
     const userCountry = countryList.find((country) => country.code === countryCode);
     if (userCountry?.code !== '' && autoFilledCountry.value === '') {
       setAutoFilledCountry({ value: userCountry?.code, displayText: userCountry?.name });
       // set formData state for auto populated country field to pass into payload
-      setFormData({ ...formData, country: userCountry?.code });
```

Comment:
Why we need to remove formData don't you think it will create issue?

Reply:
By removing it this will not create any issue. We are only setting the `country` field on the component renter, means only country field auto-populated not other fields. So we are removing it from here because it's dependency in `useEffect` was rerandring the component which was creating an issue.